dotnet-statsig/src/Statsig/Server/Evaluation/Helpers.cs

```diff
@@ -160,10 +160,12 @@ private static string NormalizeVersionString(string version)
             {
                 normalized = version.Substring(0, hyphenIndex);
             }
-            if (int.TryParse(normalized, out _))
+            var components = new List<string>(normalized.Split('.'));
+            while (components.Count < 4) 
```

Comment:
technically, node has no requirement for there to be 4 pieces to a version.  It will pad arbitrarily if one version string is longer than the other.

So 1.0 vs 1.0.0.0.0.0.0 would work in node, but not here.  should be easy enough to support that actually

Reply:
Dotnet's version only has 4 components (Major, Minor, Build, Revision).  What we're trying to fix is that in Dotnet, 

1.0.0 > 1.0, which is different from what Node does.  

Interestingly, 1.0 vs 1.0.0.0.0.0 would actually work here.

Reply:
Yeah I get that this fixes some cases.  And good point, I suppose that 1.0. vs 1.0.0.0 would work. Would longer ones work?  1.0.0.0.1 vs 1.0.0.0 would be the same then?

I guess my point was - this is definitely an improvement, but it does not match the node sdk version check as the PR title indicates.

its not about how dotnet does versioning, its about how customers pass in version strings.  If we dont need to support more than M.M.B.R, this is probably okay too